rust/core/src/automatic_branching.rs

```diff
@@ -340,11 +369,12 @@ mod tests {
 
     const MODEL_OF_INTEREST: &str = "shifts_summary";
     /// get_hash_derive_model_hash is a helper function for testing the derive_model_hash function.
-    async fn get_hash_derive_model_hash(fs: &impl FileSystem) -> String {
+    async fn get_hash_derive_model_hash(fs: &impl FileSystem, model_of_interset: Option<&str>) -> String {
```

Comment:
nit: I am not sure this abstraction is hugely useful, it makes the abstraction much more complicated for quite a slew of very simple test cases. Wonder if copy paste is better here just for the simplicity.

Reply:
If so I would document `model_of_interest` a little more explicitly as being an override over a default. But again I don't know if the abstraction is worth it.